File: python/263.丑数.py
#
# @lc app=leetcode.cn id=263 lang=python
#
# [263] 丑数
#
# 解法1(T96% S50%): 这个数不停除2，除干净之后再不停除3，再不停除5，如果最后没了(剩1)就证明是丑数
#   注意正整数的条件

# @lc code=start
class Solution(object):
    def isUgly(self, n):
        """
        :type n: int
        :rtype: bool
        """
        if n < 1:
            return False
        elif n < 7:
            return True

        for item in [2, 3, 5]:
            while n % item == 0:
                n //= item

        if n == 1:
            return True
        return False
# @lc code=end

File: javascript/263.丑数.js
/*
 * @lc app=leetcode.cn id=263 lang=javascript
 *
 * [263] 丑数
 */

// @lc code=start
/**
 * @param {number} n
 * @return {boolean}
 */
var isUgly = function(n) {
    while (n>=5 && n%5==0) {
        n /= 5;
    }
    while (n>=3 && n%3==0) {
        n /= 3;
    }
    while (n>=2 && n%2==0) {
        n /= 2;
    }

    return n==1;
};
// @lc code=end

